src/sync.rs

```diff
@@ -1,15 +1,81 @@
+#[cfg(all(not(feature = "std"), loom))]
+compile_error!("loom requires the standard library");
+
+//loom support
+
 #[cfg(loom)]
 pub(crate) mod atomic {
-    pub(crate) use loom::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
+    pub(crate) use loom::sync::atomic::{
+        fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize, Ordering,
+    };
 }
-
 #[cfg(loom)]
 pub(crate) use loom::thread::yield_now;
 
-#[cfg(not(loom))]
+//std imports
+
+#[cfg(all(not(loom), feature = "std"))]
 pub(crate) mod atomic {
-    pub(crate) use std::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
+    pub(crate) use std::sync::atomic::{
+        fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize, Ordering,
+    };
 }
-
-#[cfg(not(loom))]
+#[cfg(all(not(loom), feature = "std"))]
 pub(crate) use std::thread::yield_now;
+
+//both
+
+#[cfg(any(loom, feature = "std"))]
+pub(crate) mod marker {
+    pub(crate) use std::marker::PhantomData;
+}
+#[cfg(any(loom, feature = "std"))]
+pub(crate) mod ptr {
+    pub(crate) use std::ptr::{null, null_mut, NonNull, drop_in_place};
+}
+#[cfg(any(loom, feature = "std"))]
+pub(crate) mod mem {
+    pub(crate) use std::mem::transmute;
+}
```

Comment:
I'm not sure what the need for all these `pub(crate) use`s is? We should just always import these as `core::`, as that's available even if `std` is in use.